src/index.js

```diff
@@ -195,69 +171,48 @@ export default class HyperScreen extends React.Component {
   /**
    * Performs a full load of the screen.
    */
-  load = () => {
+  load = async () => {
     const { params, key: routeKey } = this.getNavigationState(this.props);
-    const { delay } = params;
-    const url = this.state.url;
 
-    const fetchPromise = () => this.props.fetch(url, { headers: getHyperviewHeaders() })
-      .then((response) => {
-        if (!response.ok) {
-          throw Error(response.statusText);
-        }
-        return response.text();
-      })
-      .then((responseText) => {
-        if (typeof this.props.onParseBefore === 'function') {
-          this.props.onParseBefore(url);
-        }
-        let doc = this.parser.parseFromString(responseText);
-        let error = false;
-        const stylesheets = Stylesheets.createStylesheets(doc);
-        this.navigation.setRouteKey(url, routeKey);
-
-        // Make sure the XML has the required elements: <doc>, <screen>, <body>.
-        const docElement = getFirstTag(doc, 'doc');
-        if (!docElement) {
-          console.error(`No <doc> tag found in the response from ${url}.`);
-          doc = null;
-          error = true;
-        } else {
-          const screenElement = getFirstTag(docElement, 'screen');
-          if (!screenElement) {
-            console.error(`No <screen> tag found in the <doc> tag from ${url}.`);
-            doc = null;
-            error = true;
-          } else {
-            const bodyElement = getFirstTag(screenElement, 'body');
-            if (!bodyElement) {
-              console.error(`No <body> tag found in the <screen> tag from ${url}.`);
-              doc = null;
-              error = true;
-            }
-          }
-        }
+    try {
+      if (params.delay) {
+        await later(parseInt(params.delay, 10));
+      }
 
-        this.setState({
-          doc,
-          styles: stylesheets,
-          error,
-        });
-        if (typeof this.props.onParseAfter === 'function') {
-          this.props.onParseAfter(url);
-        }
-      })
-      .catch((reason) => {
-        this.setState({
-          error: true,
-        });
-        throw reason;
+      const url = this.state.url;
+      const doc = await this.parser.load(url);
+
+      // Make sure the XML has the required elements: <doc>, <screen>, <body>.
+      const docElement = getFirstTag(doc, 'doc');
+      if (!docElement) {
+        throw new Error(`No <doc> tag found in the response from ${url}.`);
```

Comment:
You'd created specific error types elsewhere - why not for these? Seems like these could be abstracted into some kind of `ParserError` that takes `elementName` and `sourceUrl` params. Also, maybe this validation functionality could be integrated directly into `getFirstTag`?

Reply:
I deliberately left these as they are (they already exist even if the diff shows them as new code) as this seemed out of scope for the current refactor. As mentioned in the PR description, this file `src/index.js` is explicitly ignored by our linting tools as there are too many existing issues, so I'd rather limit as much as possible new code added to this file.